src/dnf-plugins/product-id.py

```diff
@@ -63,8 +64,14 @@ def transaction(self):
 
 
 class DnfProductManager(ProductManager):
-    def __init__(self, base):
-        self.base = base
+    def __init__(self):
+        # We create own base object to get fresh list of installed packages
+        # and to get list of available packages after installing/removing,
+        # because base object from dnf.Plugin provides incomplete and obsolete data.
+        self.base = dnf.base.Base()
+        self.base.read_all_repos()
+        self.base.fill_sack(load_system_repo=True, load_available_repos=True)
```

Comment:
I don't think this is good idea. Using fill_sack during dnf run as a subprocess with new base object could result in unpredictable crashes. Ok, they are predictable, and expected. 
Additionally please use only functions marked as API. Using of private functions or object, attributes, or even function not marked as API can lead in problems after upgrades. For example the function _download_productid() will not work with dnf-3.4

Reply:
@j-mracek Thanks for review, but can you give me any hint, how can I get fresh list of installed packages and available packages? We really need it, because list of installed packages does not contain information about corresponding repositories.

Good point about: `_download_productid()`. We will try to replace private/protected methods/attributes with something marked as API.